### PR TITLE
Allow divide_dim to divide to non-literal expressions

### PR DESCRIPTION
**Pull request by SamirDroubi**

### CODE REVIEW

File: src/exo/API_scheduling.py [hunk 1]
```diff
@@ -1372,8 +1372,6 @@ def divide_dim(proc, alloc_cursor, dim_idx, quotient):
         `x : R[n, 3, 4, m]`
         `x[i, j / 4, j % 4, k] = ...`
     """
-    if quotient == 1:
```
Comment (yamaguchi1024) [inline src/exo/API_scheduling.py:1375]: Can you change the docstring above "This limited implementation of `divide_dim` requires that the dimension being divided is constant itself." ?

File: src/exo/LoopIR_scheduling.py [hunk 2]
```diff
@@ -728,25 +768,10 @@ def ceildiv(lhs, rhs):
     elif tail_strategy in ["cut", "cut_and_guard"]:
         outer_hi = szop("/", N, inner_hi)  # floor div
     elif tail_strategy == "perfect":
-        if not isinstance(N, LoopIR.Const):
```
Comment (yamaguchi1024) [inline src/exo/LoopIR_scheduling.py:731]: This is great!

File: src/exo/LoopIR_scheduling.py [hunk 3]
```diff
@@ -303,6 +321,28 @@ def move_back(c):
...
+# --------------------------------------------------------------------------- #
+# IR Building Helpers
+
+
+def divide_expr(e, quot):
+    assert isinstance(e, LoopIR.expr)
+    if isinstance(quot, int):
+        quot_int = quot
+        quot_ir = LoopIR.Const(quot, e.type, e.srcinfo)
+    elif isinstance(quot, LoopIR.expr):
```
Comment (yamaguchi1024) [inline src/exo/LoopIR_scheduling.py:334]: This should be `elif isinstance(quot, LoopIR.Const):`

**Review by yamaguchi1024 — CHANGES_REQUESTED** [review]
lgtm as long as the docstring and the `elif isinstance(quot, LoopIR.Const):` thing is fixed

File: src/exo/API_scheduling.py (hunk 1 above)
Comment (SamirDroubi) [reply src/exo/API_scheduling.py:1375]: Done!

File: src/exo/LoopIR_scheduling.py (hunk 3 above)
Comment (SamirDroubi) [reply src/exo/LoopIR_scheduling.py]: Good catch!

**Review by yamaguchi1024 — APPROVED** [review]

**Pull request merged**